# Profitable App Profiles for the App Store and Google Play Markets
#### The aim for this project is to build a mobile app profile that is profitable in both the App Store and Play Store.  We're working as data analysts for a company that builds Android and iOS mobile apps.  The company only builds apps that are free to download and install, and the main source of revenue consists of in-app ads.  This means revenue for any given app is mostly influenced by the number of users who use the app — the more users that see and engage with the ads, the better. The goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

#### Collecting data for millions of apps in the app store and play store will take significant amounts of resources, so analyzing a sample of the data will be more feasible.  Luckily, there are datasets online that contain about 10,000 apps and 7,000 apps in the play store, and app store, respectively.

#### https://www.kaggle.com/lava18/google-play-store-apps
#### https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps

#### Let's start with opening the data set:

In [83]:
from csv import reader

open_file = open('AppleStore.csv')
read_file = reader(open_file)
app_store = list(read_file)
app_store_header = app_store[0]
app_store = app_store[1:]

open_file = open('googleplaystore.csv')
read_file = reader(open_file)
play_store = list(read_file)
play_store_header = play_store[0]
play_store = play_store[1:]

In [84]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

        
        
print('______________Apple App Store Data______________ ')  
print('\n')
print(app_store_header)
print('\n')
explore_data(app_store, 0, 3, True)


print('\n')
print('______________Google Play Store Data______________ ')
print('\n')
print(play_store_header)
print('\n')
explore_data(play_store, 0, 3, True)

______________Apple App Store Data______________ 


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


______________Google Play Store Data______________ 


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid

## Cleaning the Data

#### The Google Play data set has a dedicated discussion section, and we can see that one of the discussions describes an error for a certain row (https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015)
#### In the link above, Entry 10472 in the play_store dataset has an error.  Let's verify that below

In [85]:
print(play_store_header)
print('\n')

#correct data
print(play_store[1])
print('\n')

#incorrect data
print(play_store[10472])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


#### We can see that the incorrect data does not have a category column.  As a result we will delete this row.

In [86]:
print(len(play_store))

#do not run this more than once.. 
del(play_store[10472])
#comment this ^ line out after running once, read cell above to see why.

print('\n')
print(len(play_store))

10841


10840


#### If you explore the Google Play data set long enough or look at the discussions section, you'll notice some apps have duplicate entries.  Lets take a look at how many duplicate entries there are and print out a few

In [87]:
apps_duplicate = []
apps_unique = []

for each_app in play_store:
    name = each_app[0]
    if name in apps_unique:
        apps_duplicate.append(name)
    else:
        apps_unique.append(name)

        
print('Number of duplicate apps: ', len(apps_duplicate))
print('\n')
print('Names of some duplicate apps: ', apps_duplicate[:15])

Number of duplicate apps:  1181


Names of some duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


#### Before deleting the duplicate apps, we can see that the duplicate data has different numbers of reviews (Fourth entry in the list).  We can infer that the higher number of reviews within duplicate apps, the more recent the data should be. 

In [88]:
for each_app in play_store:
    name = each_app[0]
    if name == 'Google Ads':
        print(each_app)

['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29331', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


#### To prepare for the removal process, we build a dictionary with a key displaying the unique app name and the value is the higher number of reviews of that app.  
#### To verify that the data is correct, we print the difference of the length of raw data and its duplicates

In [89]:
reviews_max = {}
for each in play_store:
    name = each[0]
    n_reviews = float(each[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
    
print('Expected Length: ', len(play_store) - 1181)
print('Actual Length: ', len(reviews_max))

Expected Length:  9659
Actual Length:  9659


#### Now that we have the max reviews of each unique app in a dictionary, we can use it to clean out the duplicates in the original dataset and verify by running our explore_data function and comparing the row to length

In [90]:
play_store_clean = []
already_added = []

for each in play_store:
    name = each[0]
    num_reviews = float(each[3])
    
    if (reviews_max[name] == num_reviews) and (name not in already_added):
        play_store_clean.append(each)
        already_added.append(name)

explore_data(play_store_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


#### Now that duplicate apps in the google play store are removed, let's check the apple ios store for duplicates

In [91]:
appstore_duplicate = []
appstore_unique = []

for each_app in app_store:
    name = each_app[0]
    if name in apps_unique:
        appstore_duplicate.append(name)
    else:
        appstore_unique.append(name)
        
print('Number of duplicate apps: ', len(appstore_duplicate))
print('\n')
print('Names of some duplicate apps: ', appstore_duplicate[:15])

Number of duplicate apps:  0


Names of some duplicate apps:  []


## Removing Non-English Apps

#### We're not interested in keeping non-english apps, so we'll remove them.  All these characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it, and we can take advantage of that to build a function that checks an app name and tells us whether it contains non-ASCII characters.

In [92]:
def is_english(string):
    for char in string:
        if ord(char) > 127:
            return False
    return True

print(is_english('Hello'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Instachat 😜'))
print(is_english('Docs To Go™ Free Office Suite'))

True
False
False
False


#### The function works fine, but emojis and other symbols fall outside of the ASCII range we set.  To minimize data loss, we will only remove an app where its name has more than 3 non-ASCII characters. Although not perfect, but only very few non-english apps will pass through this filter, and is good enough at this point.

In [93]:
def is_english(string):
    notascii = 0
    for char in string:
        if ord(char) > 127:
            notascii += 1
            
    if notascii > 3:
        return False
    else:
        return True
    
print(is_english('Hello'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Instachat 😜'))
print(is_english('Docs To Go™ Free Office Suite'))

True
False
True
True


In [94]:
play_store_english = []
app_store_english = []

for each in play_store_clean:
    if is_english(each[0]) == True:
        play_store_english.append(each)
        
for each in app_store:
    if is_english(each[0]) == True:
        app_store_english.append(each)

explore_data(play_store_english, 0, 3, True)
print('\n')
explore_data(app_store_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

In [95]:
play_store_final = []
app_store_final = []

for each in play_store_english:
    price = each[7]
    if price == '0':
        play_store_final.append(each)

        
for each in app_store_english:
    price = each[4]
    if price == '0.0':
        app_store_final.append(each)
        
explore_data(play_store_final, 0, 3, True)
print('\n')
explore_data(app_store_final, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

#### As mentioned earlier, our aim is to determine the kind of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.
#### To minimize risk / overhead, our validation strategy for an app idea consists of three steps:
#### 1. Build a minimal version Android version of the app, and add it to Google Play <br> 2. If the app has a good response from users, develop further <br> 3. If the app is profitable after six months, build an iOS version and add it to the App Store
#### Since the end goal is to have the app in both App Store and Play Store, we will develop an app profile that is successful in both market. We will begin the analysis with getting a sense of what the most common genre's are in the market.

In [96]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
    
    return table_percentages

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], " : ", entry[0])

In [97]:
display_table(app_store_final, 11) #prime_genre column

Games  :  55.64595660749507
Entertainment  :  8.234714003944774
Photo & Video  :  4.117357001972387
Social Networking  :  3.5256410256410255
Education  :  3.2544378698224854
Shopping  :  2.983234714003945
Utilities  :  2.687376725838264
Lifestyle  :  2.3175542406311638
Finance  :  2.0710059171597637
Sports  :  1.947731755424063
Health & Fitness  :  1.8737672583826428
Music  :  1.6518737672583828
Book  :  1.6272189349112427
Productivity  :  1.5285996055226825
News  :  1.4299802761341223
Travel  :  1.3806706114398422
Food & Drink  :  1.0601577909270217
Weather  :  0.7642998027613412
Reference  :  0.4930966469428008
Navigation  :  0.4930966469428008
Business  :  0.4930966469428008
Catalogs  :  0.22189349112426035
Medical  :  0.19723865877712032


#### From examing the prime_genre column in the app store, we can see that the market space is dominated by fun apps such as games, entertainment, photo & video, and social networking.  Practical apps are much more rare in this instance, but one thing to consider is that the greater number of apps does not imply that there are a greater number of users.

In [98]:
display_table(play_store_final, 1) # Category Column

FAMILY  :  18.907942238267147
GAME  :  9.724729241877256
TOOLS  :  8.461191335740072
BUSINESS  :  4.591606498194946
LIFESTYLE  :  3.9034296028880866
PRODUCTIVITY  :  3.892148014440433
FINANCE  :  3.7003610108303246
MEDICAL  :  3.531137184115524
SPORTS  :  3.395758122743682
PERSONALIZATION  :  3.3167870036101084
COMMUNICATION  :  3.2378158844765346
HEALTH_AND_FITNESS  :  3.0798736462093865
PHOTOGRAPHY  :  2.944494584837545
NEWS_AND_MAGAZINES  :  2.7978339350180503
SOCIAL  :  2.6624548736462095
TRAVEL_AND_LOCAL  :  2.33528880866426
SHOPPING  :  2.2450361010830324
BOOKS_AND_REFERENCE  :  2.1435018050541514
DATING  :  1.861462093862816
VIDEO_PLAYERS  :  1.7937725631768955
MAPS_AND_NAVIGATION  :  1.3989169675090252
FOOD_AND_DRINK  :  1.2409747292418771
EDUCATION  :  1.1620036101083033
ENTERTAINMENT  :  0.9589350180505415
LIBRARIES_AND_DEMO  :  0.9363718411552346
AUTO_AND_VEHICLES  :  0.9250902527075812
HOUSE_AND_HOME  :  0.8235559566787004
WEATHER  :  0.8009927797833934
EVENTS  :  0.7107400

#### We can see that the landscape for google play is much different.  With Family type apps at the top, followed by games, and tools.  Today, the family category has been revised to Kids and other subcategories.  When looking into the Kids categories, we will see that it consists of mostly games.
#### Practical apps seem to have a better presence in the play store vs the app store.  As shown in the Genre column below

In [99]:
display_table(play_store_final, -4) #Genre Column

Tools  :  8.449909747292418
Entertainment  :  6.069494584837545
Education  :  5.347472924187725
Business  :  4.591606498194946
Productivity  :  3.892148014440433
Lifestyle  :  3.892148014440433
Finance  :  3.7003610108303246
Medical  :  3.531137184115524
Sports  :  3.463447653429603
Personalization  :  3.3167870036101084
Communication  :  3.2378158844765346
Action  :  3.1024368231046933
Health & Fitness  :  3.0798736462093865
Photography  :  2.944494584837545
News & Magazines  :  2.7978339350180503
Social  :  2.6624548736462095
Travel & Local  :  2.3240072202166067
Shopping  :  2.2450361010830324
Books & Reference  :  2.1435018050541514
Simulation  :  2.0419675090252705
Dating  :  1.861462093862816
Arcade  :  1.8501805054151623
Video Players & Editors  :  1.7712093862815883
Casual  :  1.7599277978339352
Maps & Navigation  :  1.3989169675090252
Food & Drink  :  1.2409747292418771
Puzzle  :  1.128158844765343
Racing  :  0.9927797833935018
Role Playing  :  0.9363718411552346
Libraries & D

#### It isn't clear what the difference is between Genre and Category under the play store.  With Genre displaying more categories, we will move forward with the Category column instead as we are looking at the bigger picture.

## Most popular apps by Genre
#### One way to find the most popular genre's is to calculate the number of installs for each app genre.  
#### In the Google Play dataset, we can find this information in the installs column.  In the Apple Apps Store, the information is missing, so as a work around we will take the average total number of user ratings by genre, shown below

In [100]:
unique_app_genres = freq_table(app_store_final, 11)

for genre in unique_app_genres:
    total = 0
    len_genre = 0
    
    for app in app_store_final:
        genre_app = app[11]
        if genre_app == genre:
            total += float(app[5])
            len_genre += 1
    
    print(genre, ":", total / len_genre)


Social Networking : 53078.195804195806
Photo & Video : 27249.892215568863
Games : 18924.68896765618
Music : 56482.02985074627
Reference : 67447.9
Health & Fitness : 19952.315789473683
Weather : 47220.93548387097
Utilities : 14010.100917431193
Travel : 20216.01785714286
Shopping : 18746.677685950413
News : 15892.724137931034
Navigation : 25972.05
Lifestyle : 8978.308510638299
Entertainment : 10822.961077844311
Food & Drink : 20179.093023255813
Sports : 20128.974683544304
Book : 8498.333333333334
Finance : 13522.261904761905
Education : 6266.333333333333
Productivity : 19053.887096774193
Business : 6367.8
Catalogs : 1779.5555555555557
Medical : 459.75


#### On average, reference apps have the highest number of user reviews, followed by music, and social networking.  However, this is most likely due to a few big name apps in their respective categories (Facebook, Spotify, Soundcloud).  Below is a list of each app in their categories with # of reviews.

In [101]:
for app in app_store_final:
    if app[11] == 'Reference':
        print(app[1], ':', app[5]) #print name and # of ratings

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
彩库宝典-【官方版】 : 0
Jishokun-Japanese English Dictionary & Translator : 0
無料で音楽や写真・カメラの裏技アプリ for iPhone7 : 0


In [102]:
for app in app_store_final:
    if app[11] == 'Music':
        print(app[1], ':', app[5]) #print name and # of ratings

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

In [103]:
for app in app_store_final:
    if app[11] == 'Social Networking':
        print(app[1], ':', app[5]) #print name and # of ratings

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

#### With Social networking and Music heavily skewed towards the few giants.. turning to reference seems to show potential.  

#### Since the app store is dominated by fun apps, creating a practical app has a better chance of standing out among the abundance of apps in the apps store.  Something that could be done is to take a popular book and turn it into an app, and add different features to compliment the raw version of the book (Audio version...Daily Quotes... Dictionary / Index)


## Most Popular Apps by Genre on Google Play

#### In the Google dataset, there is a column that indicates the number of times an app has been installed.  However, the values are not accurate as they are open ended.

In [104]:
display_table(play_store_final, 5)

1,000,000+  :  15.726534296028879
100,000+  :  11.552346570397113
10,000,000+  :  10.548285198555957
10,000+  :  10.198555956678701
1,000+  :  8.393501805054152
100+  :  6.915613718411552
5,000,000+  :  6.825361010830325
500,000+  :  5.561823104693141
50,000+  :  4.7721119133574
5,000+  :  4.512635379061372
10+  :  3.5424187725631766
500+  :  3.2490974729241873
50,000,000+  :  2.3014440433213
100,000,000+  :  2.1322202166064983
50+  :  1.917870036101083
5+  :  0.78971119133574
1+  :  0.5076714801444043
500,000,000+  :  0.2707581227436823
1,000,000,000+  :  0.22563176895306858
0+  :  0.04512635379061372
0  :  0.01128158844765343


#### In our case, we won't know whether an app that has 100,000+ installs actually has 100,001 installs, or 250,000 installs. Luckily, we only want to get an idea which app genres attract the most users, so we won't need an exact value of the number of users.

#### The next course of action is to leave the numbers as they are, so an app with 100,000+ installs, will just be 100,000 installs.  To do this, we have to remove the comma's and +'s in the string, and then convert to a float in order to do our calculations.  See below:

In [105]:
unique_app_genres = freq_table(play_store_final, 1)

for genre in unique_app_genres:
    total = 0
    len_category = 0
    
    for app in play_store_final:
        genre_app = app[1]
        if genre_app == genre:
            installs = app[5].replace('+', '')
            installs = installs.replace(',', '')
            total += float(installs)
            len_category += 1
    
    print(genre, ":", total / len_category)


ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

#### Communication apps have the most installs at 38,456,119. This average skewed by large name apps, similar to the Apple iOS app store.

In [106]:
for app in play_store_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

#### If we remove all apps under communication with over 100 million installs, the average would reduce greatly (about 10 times)

In [107]:
under_100m = []

for app in play_store_final:
    installs = app[5]
    installs = installs.replace(',', '')
    installs = installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(installs) < 100000000):
        under_100m.append(float(installs))
        
sum(under_100m) / len(under_100m)

3603485.3884615386

#### The trend continues with social, video, and game genre apps

In [108]:
for app in play_store_final:
    if app[1] == 'SOCIAL' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+


In [109]:
for app in play_store_final:
    if app[1] == 'VIDEO_PLAYERS' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

YouTube : 1,000,000,000+
Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Google Play Movies & TV : 1,000,000,000+
MX Player : 500,000,000+
Dubsmash : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


In [110]:
for app in play_store_final:
    if app[1] == 'GAME' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Sonic Dash : 100,000,000+
PAC-MAN : 100,000,000+
Roll the Ball® - slide puzzle : 100,000,000+
Piano Tiles 2™ : 100,000,000+
Pokémon GO : 100,000,000+
Extreme Car Driving Simulator : 100,000,000+
Trivia Crack : 100,000,000+
Angry Birds 2 : 100,000,000+
Candy Crush Saga : 500,000,000+
8 Ball Pool : 100,000,000+
Subway Surfers : 1,000,000,000+
Candy Crush Soda Saga : 100,000,000+
Clash Royale : 100,000,000+
Clash of Clans : 100,000,000+
Plants vs. Zombies FREE : 100,000,000+
Pou : 500,000,000+
Flow Free : 100,000,000+
My Talking Angela : 100,000,000+
slither.io : 100,000,000+
Cooking Fever : 100,000,000+
Yes day : 100,000,000+
Score! Hero : 100,000,000+
Dream League Soccer 2018 : 100,000,000+
My Talking Tom : 500,000,000+
Sniper 3D Gun Shooter: Free Shooting Games - FPS : 100,000,000+
Zombie Tsunami : 100,000,000+
Helix Jump : 100,000,000+
Crossy Road : 100,000,000+
Temple Run 2 : 500,000,000+
Talking Tom Gold Run : 100,000,000+
Agar.io : 100,000,000+
Bus Rush: Subway Edition : 100,000,00

#### Given that our main concern is finding an app that is profitable in both the play store and app store.  We can take a dive further into the books and reference category, since we have shown that references under the app store would be a good profile.

In [111]:
for each in play_store_final:
    if each[1] == 'BOOKS_AND_REFERENCE':
        print(each[0], ":", each[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

#### We can see here that the number of large name apps is fairly lower than the top genres, generating a much smaller bias towards the average.

In [112]:
for app in play_store_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


#### The BOOKS_AND_REFERENCE category seems to consist of apps for dictionaries, libraries, processing and reading ebooks.  Building an app off of that will see significant competition.  There are also some apps built around the Quran, indicating that building an app off of a popular and/or more recent book could be profitable in both the Apple and Android market.  Since the market already a lot of libraries, including features such as an audio book, daily quotes, etc. will help the app to stand out.

## Conclusion

#### In this project, we analyzed data about app stores in the Google Play and Apple iOS platform to create an app profile that would be potentially profitable in both markets.  

#### Our findings indicate that taking a more recent and popular book and turning it into an app would be profitable in both markets.  In order to stand out from the existing competition, features such as an audio version, daily quotes, etc. should be included to make the app unique.